yaksh/templates/yaksh/add_question.html

```diff
@@ -1,77 +1,121 @@
+            <input type="checkbox" name="extract" value="{{file.id}}" >{% if file.extract %} dont extract{% else %}extract{% endif %}</input>&nbsp;&nbsp;
+            <input type="checkbox" name="hide" value="{{file.id}}" >{% if file.hide %} show{% else %}hide{% endif %}</input>
+            <a href="{{file.file.url}}">{{ file.file.name }}</a>
+            <br>
+          {% endfor %}
+        {% endif %}
+      </table>
+    </center>
+    {% for formset in formsets %}
+    <div class="form-group">
+        {{ formset.management_form }}
 
-      {% for form in formset %}
-        <div class="link-formset well">
-            {{ form.as_p }}
-        </div>
-      {% endfor %}
+        {% for form in formset %}
+          <div class="link-formset well">
+              {{ form.as_p }}
+          </div>
+        {% endfor %}
 
-  </div>
-  {% endfor %}
-  <p><label for="case_type">Add Test Case:</label> <select id="case_type" name="case_type" onchange="frm.submit()">
-      <option value="" selected="selected">---------</option>
-      <option value="standardtestcase">Standard </option>
-      <option value="stdiobasedtestcase">StdIO </option>
-      <option value="mcqtestcase">MCQ/MCC </option>
-      <option value="hooktestcase">Hook </option>
-      <option value="integertestcase">Integer </option>
-      <option value="stringtestcase"> String </option>
-      <option value="floattestcase"> Float </option>
-      <option value="arrangetestcase">Arrange options </option>
-  </select></p>
-  <center>
-    <button class="btn btn-success" type="submit" name="save_question">Save</button>
-    <button class="btn btn-info" type="button" name="button" onClick='location.replace("{{URL_ROOT}}/exam/manage/questions/");'>Back to Questions</button>
-    <button class="btn btn-danger" type="submit" name="delete_files">Delete Selected Files</button>
-  </center>
-</form>
+    </div>
+    {% endfor %}
+      <div class="form-group">
+        {% if question %}    
+          <label for="case_type">Add Test Case:</label>
+            {% if question.type == "code" %}
+              {% if qform.instance.language == "bash" %}
+                <select id="case_type" class="form-control w-auto" name="case_type" onchange="frm.submit()">
+                  <option value="" selected="selected">---------</option>
+                  <option value="standardtestcase">Standard </option>
+                  <option value="stdiobasedtestcase">StdIO </option>
+                  <option value="hooktestcase">Hook </option>
+                </select>
+              {% elif qform.instance.language == "scilab" %}
+                <select id="case_type" class="form-control w-auto" name="case_type" onchange="frm.submit()">
+                  <option value="" selected="selected">---------</option>
+                  <option value="standardtestcase">Standard </option>
+                  <option value="hooktestcase">Hook </option>
+                </select>
```

Comment:
Instead of having multiple if-else to show the test case options based on language selection, it would be better to create a dictionary that will contain available code evaluators for a specific language. For e.g. {'python': ['standardtestcase', 'stdiotestcase', 'hooktestcase']}.